cou/utils/openstack.py

```diff
@@ -330,3 +330,18 @@ def lookup(cls, component: str, version: str) -> List[OpenStackRelease]:
             if version in version_range:
                 compatible_os_releases.append(OpenStackRelease(openstack_release))
         return compatible_os_releases
+
+    @classmethod
+    def charm_supported(cls, component: str) -> bool:
```

Comment:
this function is so similar to `lookup`... could we get by without it? `AppFactory.create` could simply interpret an empty list returned by lookup as 'not supported'

Reply:
That's right, you could still get an empty list from lookup even if the component name is in the table, I had not considered that. I also had not considered that the lookup function would not have worked unless you also pass a version. So... in retrospect my advice was not particularly good.

Now, your new solution works, but relies on side-effects of the lookup function, which now does double duty as both a get_compatible_versions and is_openstack_component... and makes me realize that the name `lookup` wasn't all that accurate in the first place.

Here's a different proposal that still remains DRY:
* keep the lookup function as an actual (possibly private) lookup function: it'd simply set up `cls._OPENSTACK_LOOKUP`, and return whatever `cls._OPENSTACK_LOOKUP.get(component)` finds (possibly an exception)
* write a "find compatible versions" function that uses the lookup (i.e. the old `lookup()`)
* write a separate "is this charm supported?" function that uses the lookup and returns a boolean

Reply:
I ended up changing the method `lookup` to ` find_compatible_versions` and creating a new one named `is_charm_supported` that is used in the `find_compatible_versions` and populates `_OPENSTACK_LOOKUP` if necessary.